### PR TITLE
GUI: Replace welcome dialog with welcome tab, enable closable tabs

### LINKED ISSUE #151: GUI: landing page instead of splash window
## Expected Behavior
Augment the existing welcome splash with additional information / options

This could possibly be a landing splash page, rather than an independent widget that pops up.  Maybe something similar to this? 🤔 

image

Might have the following:
```
Welcome to atef (8f logo)

New
   -> Passive configuration
   -> Active configuration
Open an existing file
(Recent file listing)
Close
Exit
```

## Current Behavior
A simple, though very operational welcome splash widget, where "new" opens up another window

image

## Context
Discussion sparked by Ken in #150

## Your Environment
pcds-5.6.0

### PR DESCRIPTION
**Pull request by tangkong**
## Description
- Replace the welcome dialog with a landing tab, closes #151 
- Set up close buttons, and move run-toggle to reveal it

## Motivation and Context
I grow tired of users losing the welcome splash in their array of 6 monitors and complaining atef is frozen.

## How Has This Been Tested?
Interactively

## Where Has This Been Documented?
This PR

image

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Code has been checked for threading issues (no blocking tasks in GUI thread)
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran ``docs/pre-release-notes.sh`` and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

File: atef/widgets/config/window.py [hunk 1]
```diff
@@ -72,24 +78,41 @@ def __init__(self, *args, show_welcome: bool = True, **kwargs):
...
+            partial(self.new_file, checkout_type='passive')
+        )
+        widget.new_active_button.clicked.connect(
+            partial(self.new_file, checkout_type='active')
+        )
+        widget.sample_active_button.clicked.connect(partial(
+            self.open_file, filename=TEST_CONFIG_PATH / 'active_test.json'
+        ))
+
+        widget.sample_active_button.clicked.connect(partial(
```
Comment (klauer) [inline atef/widgets/config/window.py:106]:
```suggestion
        widget.sample_passive_button.clicked.connect(partial(
```
Found through local testing

**Review by klauer — APPROVED** [review]
This looks great and works well!

A couple more things to consider after this:
1. "File -> new" could probably be reworked using this dialog
2. Now that it's easier to open tabs, I notice how poorly they use the space allotted for them. Any idea on how to fix it without spending too much effort?

image

File: atef/widgets/config/window.py (hunk 1 above)
Comment (tangkong) [reply atef/widgets/config/window.py:106]: I'm just keeping you on your toes 😉

**Comment by tangkong** [comment]
Interesting spacing...  I can't reproduce that immediately.  You're opening the atef gui on your mac?
on psbuild: 
image

on local mac: (opening more tabs forcefully expands the GUI window)
image

**Comment by klauer** [comment]
> Interesting spacing... I can't reproduce that immediately. You're opening the atef gui on your mac?

Yes, locally on my mac. 
Given that you can't reproduce it, I guess it's a me-problem and not worth investigating

**Comment by tangkong** [comment]
I think I managed to lock down the tab buttons/text truncation.  Mind taking another look?

**Comment by klauer** [comment]
Ah-ha, you did it! And you caught the same thing I was thinking about the 'close' button no longer existing. Nice:

image

**Comment by tangkong** [comment]
We can never really take the contents of the "default" stylesheet for granted I guess 😆

**Comment by tangkong** [comment]
Also merging this, thanks for the review 👍

**Comment by ZLLentz** [comment]
Whoa, great improvement here!

**Pull request merged**